x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmAuthenticateAccessTokenTypeTests.java

```diff
@@ -0,0 +1,167 @@
+            randomBoolean() // createHttpsServer
+        );
+        final JwtIssuerAndRealm jwtIssuerAndRealm = randomJwtIssuerRealmPair();
+        final User user = randomUser(jwtIssuerAndRealm.issuer());
+
+        final SecureString jwt = randomJwt(jwtIssuerAndRealm, user);
+        final SecureString clientSecret = JwtRealmInspector.getClientAuthenticationSharedSecret(jwtIssuerAndRealm.realm());
+        doMultipleAuthcAuthzAndVerifySuccess(jwtIssuerAndRealm.realm(), user, jwt, clientSecret, randomIntBetween(1, 3));
+    }
+
+    public void testAccessTokenTypeWorksWithFallbacks() throws Exception {
+        randomFallbacks();
+
+        jwtIssuerAndRealms = generateJwtIssuerRealmPairs(
+            createJwtRealmsSettingsBuilder(),
+            randomIntBetween(1, 1), // realms
+            randomIntBetween(0, 1), // authz
+            randomIntBetween(1, JwtRealmSettings.SUPPORTED_SIGNATURE_ALGORITHMS.size()), // algorithms
+            randomIntBetween(1, 3), // audiences
+            randomIntBetween(1, 3), // users
+            randomIntBetween(0, 3), // roles
+            randomIntBetween(0, 1), // jwtCacheSize
+            randomBoolean() // createHttpsServer
+        );
+        final JwtIssuerAndRealm jwtIssuerAndRealm = randomJwtIssuerRealmPair();
+        final User user = randomUser(jwtIssuerAndRealm.issuer());
+
+        final SecureString jwt2 = randomJwt(jwtIssuerAndRealm, user);
+        final SecureString clientSecret = JwtRealmInspector.getClientAuthenticationSharedSecret(jwtIssuerAndRealm.realm());
+        doMultipleAuthcAuthzAndVerifySuccess(jwtIssuerAndRealm.realm(), user, jwt2, clientSecret, randomIntBetween(1, 3));
+    }
+
+    @Override
+    protected JwtRealmSettingsBuilder createJwtRealmSettingsBuilder(JwtIssuer jwtIssuer, int authzCount, int jwtCacheSize)
+        throws Exception {
+        final JwtRealmSettingsBuilder jwtRealmSettingsBuilder = super.createJwtRealmSettingsBuilder(jwtIssuer, authzCount, jwtCacheSize);
+        final String realmName = jwtRealmSettingsBuilder.name();
+        final Settings.Builder settingsBuilder = jwtRealmSettingsBuilder.settingsBuilder();
+        settingsBuilder.put(RealmSettings.getFullSettingKey(realmName, JwtRealmSettings.TOKEN_TYPE), "access_token")
+            .putList(
+                RealmSettings.getFullSettingKey(realmName, JwtRealmSettings.ALLOWED_SUBJECTS),
+                jwtIssuer.principals.keySet().stream().toList()
+            );
+
+        if (fallbackSub != null) {
+            settingsBuilder.put(RealmSettings.getFullSettingKey(realmName, JwtRealmSettings.FALLBACK_SUB_CLAIM), fallbackSub);
+        }
+        if (fallbackAud != null) {
+            settingsBuilder.put(RealmSettings.getFullSettingKey(realmName, JwtRealmSettings.FALLBACK_AUD_CLAIM), fallbackAud);
+        }
+
+        return jwtRealmSettingsBuilder;
+    }
+
+    @Override
+    protected SecureString randomJwt(JwtIssuerAndRealm jwtIssuerAndRealm, User user) throws Exception {
```

Comment:
This method is mostly a copy/paste from JwtRealmTestCase. I think it can be simplified if we use a builder pattern for the Unsigned JWT. But I'll leave it to a future it.